docs/docs/configuring-usage-with-plugin-options.md

```diff
@@ -54,6 +54,24 @@ The following table lists possible options values and an example plugin that mak
 
 **Note**: Themes (which are a type of plugin) are able to receive options from a site's `gatsby-config` to be used in its `gatsby-config` in order to allow themes to be composed together. This is done by exporting the `gatsby-config` as a function instead of an object. You can see an example of this in the [`gatsby-theme-blog`](https://github.com/gatsbyjs/themes/tree/master/packages/gatsby-theme-blog) and [`gatsby-theme-blog-core`](https://github.com/gatsbyjs/themes/tree/master/packages/gatsby-theme-blog-core) repositories. Plugins are not capable of this functionality.
 
+## How to validate options
+
+A Gatsby plugin can define a [Joi](https://joi.dev) schema for their options using the `pluginOptionsSchema` API in `gatsby-node.js`. Gatsby validates that the options users of the plugin pass to it match the schema.
```

Comment:
It would be great to reframe this paragraph to highlight the value add of option validation. Why might plugin authors want to add option validation? Does validation make things easier for plugin users or plugin authors?

I'm imagining something like "In order to make your plugin easier to use, you can define a Joi schema to enforce data types for each plugin option. To create a schema, you can use the pluginOptionsSchema API in your gatsby-node.js file."

I don't know if that exact content is accurate, so feel free to make adjustments. My main thought is that we should add something to help users decide if this is something they should use or not.